### ISSUE TITLE
Deprecated: Implicit conversion from float to int loses precision

### ISSUE THREAD

**Issue by lorenzo-piersante** [alexcrawford/lexorank-php#6]
Starting from PHP 8.0 implicit conversion from float to int is deprecated.

From phpunit depracations report I get a lot of errors similar to the following one:
```
Deprecated: Implicit conversion from float 98.5 to int loses precision in /app/vendor/alexcrawford/lexorank-php/src/Rank.php on line 75
```

#

In ordert to fix this issue the following line (Rank.php on line 75):

```php
return chr((ord($prev) + ord($next)) / 2);
```

should be changed to:

```php
return chr((int) round((ord($prevChar) + ord($nextChar)) / 2))
```

**Comment by lorenzo-piersante** [comment]
If you agree, I can handle the PR

**Comment by Ocramius** [comment]
@lorenzo-piersante do we have an example which triggers this? Would need to be in the test suite.

**Comment by lorenzo-piersante** [comment]
@Ocramius In my Symfony project with PHP 8.1 and "phpunit/phpunit": "^8" i can reproduce the warning with the following test:

```php
public function testImplicitConversionWarning() : void
    {
        $newRank = (new Rank('az', 'b'))->get();

        self::assertIsString($newRank);
    }
```

Output:

```
PHPUnit 8.5.27 #StandWithUkraine

.                                                                   1 / 1 (100%)
Deprecated: Implicit conversion from float 97.5 to int loses precision in /app/vendor/alexcrawford/lexorank-php/src/Rank.php on line 75
```

I think we just need to upgrade phpunit to a version >= 8 to reproduce the error in this repository aswell.

**Comment by Ocramius** [comment]
Perhaps adding a CI pipeline to this project? You can take inspiration from something like https://github.com/Roave/BackwardCompatibilityCheck/blob/1ba646d3572c812c8d1e37afd66d7a8363121c9b/.github/workflows/continuous-integration.yml perhaps?

**Comment by lorenzo-piersante** [comment]
Investigating a little bit further I realized that the warning is generated in my symfony project because the "symfony/phpunit-bridge" package is installed.
In fact the warning is reproduced also by the tests in this package if I install the phpunit-bridge component in the lexorank repo. 😅

With "symfony/phpunit-bridge": "^6.1"
```bash
$ vendor/bin/simple-phpunit

PHPUnit 9.5.21 #StandWithUkraine

Testing 
.......                                                             7 / 7 (100%)

Time: 00:00.003, Memory: 6.00 MB

OK (7 tests, 7 assertions)

Remaining self deprecation notices (3)

  2x: Implicit conversion from float 97.5 to int loses precision
    1x in RankTest::testSuccessNewDigit from AlexCrawford\LexoRank\Tests
    1x in RankTest::testSuccessNewDigitMidValue from AlexCrawford\LexoRank\Tests

  1x: Implicit conversion from float 109.5 to int loses precision
    1x in RankTest::testSuccessNewDigitMidValueSpecialCase from AlexCrawford\LexoRank\Tests
```
So it seems to me that the one I reported is not a deprecation for php, but a deprecation specific to symfony.
Do you think it makes sense to install this component outside of a symfony application just to test for this warning?

The CI pipeline is a great idea, anyway I think it is not necessary to test using a matrix of php versions for this specific issue!

**Comment by Ocramius** [comment]
If you copy that pipeline, it should implicitly create the matrix according to current `composer.json`, which reduces maintenance effort later on :) 

Here's a more simplified version: https://github.com/laminas/laminas-servicemanager/blob/dc255c268124f19709f1494aa1259967d2ecab1a/.github/workflows/continuous-integration.yml#L1-L11

**Comment by Ocramius** [comment]
BTW, running PHPUnit with `convertDeprecationsToExceptions="true"` suffices :+1:

**Comment by guidobonuzzi** [comment]
Fixed by #7
